openedx/adg/lms/applications/tests/test_views.py

```diff
@@ -416,232 +436,419 @@ def test_get_redirects_without_login_for_education_experience_view():
+        user.application.background_question = TEST_BACKGROUND_QUESTION if bg_question else ''
 
-    mock_render.assert_called_once_with(
-        cover_letter_view_post_request, 'adg/lms/applications/cover_letter.html', expected_context
-    )
+    response = logged_in_client.post(reverse('application_business_line_interest'))
+    assert response.status_code == HTTP_400_BAD_REQUEST
 
 
 @pytest.mark.django_db
 @pytest.mark.parametrize(
-    'button, template',
-    [('back', 'application_education_experience'), ('submit', 'application_hub')],
-    ids=['back', 'submit']
+    'button, interest_in_business, template',
+    [
+        ('back', TEST_INTEREST_IN_BUSINESS, 'application_education_experience'),
+        ('back', '', 'application_education_experience'),
+        ('submit', TEST_INTEREST_IN_BUSINESS, 'application_hub')
+    ], ids=['back_with_interest', 'back_with_no_interest', 'submit_with_interest']
 )
-def test_post_back_or_submit_written_application_cover_letter_view(button, template, cover_letter_view_post_request):
-    # pylint: disable=protected-access
+# pylint: disable=unused-argument,protected-access
+def test_post_back_or_submit_with_required_data_business_line_interest_view(
+    button,
+    interest_in_business,
+    template,
+    business_line_interest_view_post_request,
+    prereqs_completed_business_line_interest_view
+):
     """
-    Test that the user is redirected to experience on clicking back and to application hub on submitting.
+    Tests that the user can post via `back` button with or without adding any interest in business
+    Also tests that the user can post via `submit` button with interest in business
     """
     business_line = BusinessLineFactory()
 
-    _mutable = cover_letter_view_post_request.POST._mutable
-    cover_letter_view_post_request.POST._mutable = True
-    cover_letter_view_post_request.POST['business_line'] = business_line.id
-    cover_letter_view_post_request.POST['button_click'] = button
-    cover_letter_view_post_request.POST._mutable = _mutable
-
-    response = CoverLetterView.as_view()(cover_letter_view_post_request)
+    _mutable = business_line_interest_view_post_request.POST._mutable
+    business_line_interest_view_post_request.POST._mutable = True
+    business_line_interest_view_post_request.POST['business_line'] = business_line.id
+    business_line_interest_view_post_request.POST['interest_in_business'] = interest_in_business
+    business_line_interest_view_post_request.POST['submit_or_back_clicked'] = button
+    business_line_interest_view_post_request.POST._mutable = _mutable
 
+    response = BusinessLineInterestView.as_view()(business_line_interest_view_post_request)
     assert response.get('Location') == reverse(template)
-    assert response.status_code == 302
+    assert response.status_code == HTTP_302_FOUND
+    assert business_line_interest_view_post_request.user.application.interest_in_business == interest_in_business
```

Comment:
What purpose does this assert serve?

Reply:
This basically checks if the interest_in_business field is being set correctly or not as a result of the form being submitted.